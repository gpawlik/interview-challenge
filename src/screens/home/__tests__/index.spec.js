import * as React from "react";
import renderer from "react-test-renderer";

import { HomeScreen } from "../";

jest.mock("../components/item-picker", () => ({
  ItemPicker: "ItemPicker",
}));
jest.mock("../components/menu-preview", () => ({
  MenuPreview: "MenuPreview",
}));
jest.mock("../components/summary", () => ({
  Summary: "Summary",
}));

jest.mock("../styles", () => ({
  Wrapper: "Wrapper",
  MenuSummary: "MenuSummary",
  MenuBuilder: "MenuBuilder",
}));

describe("HomeScreen", () => {
  it("renders correctly", () => {
    const component = renderer.create(<HomeScreen />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
