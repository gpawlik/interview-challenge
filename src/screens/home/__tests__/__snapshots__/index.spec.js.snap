// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<Wrapper>
  <MenuSummary>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <Summary />
      </div>
    </div>
  </MenuSummary>
  <MenuBuilder
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-4"
      >
        <ItemPicker />
      </div>
      <div
        className="col-8"
      >
        <h2>
          Menu preview
        </h2>
        <MenuPreview />
      </div>
    </div>
  </MenuBuilder>
</Wrapper>
`;
